Source/Scene/Primitive.js

```diff
@@ -867,6 +874,19 @@ define([
         return vsPick;
     };
 
+    Primitive._appendOffsetToShader = function(primitive, vertexShaderSource) {
+        if (!defined(primitive._batchTableAttributeIndices.offset)) {
+            return vertexShaderSource;
+        }
+
+        var attr = 'attribute float batchId;\nattribute float applyOffset;';
+        var modifiedShader = vertexShaderSource.replace(/attribute\s+float\s+batchId;/g, attr);
+
+        var str = 'vec4 p = czm_computePosition();\n    p = p + vec4(czm_batchTable_offset(batchId) * applyOffset, 0.0);';
```

Comment:
I think it's always defined?  At least we don't check if it's defined in any of the other shaders that use `czm_computePosition`

Reply:
Sorry, I used the wrong term. The value returned by `czm_computePosition` is always defined (it must be since it's GLSL). I meant you should use regex captures to get the name of the variable, in this case `p`.